### PR TITLE
Implement EXISTS Subquery     

### PR DESCRIPTION
**Pull request by dehowef**
Implements a naive version of EXISTS subquery that evaluates the underlying subquery.

Add regression tests related to EXISTS
TODO: Implement logic to allow the underlying subquery to constrain the queries in the outer scope.

Also fixes a minor typo.

### CODE REVIEW

File: regress/expected/cypher_subquery.out [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+										   WHERE EXISTS {(a:person)-[]->(b:pet)}
+										   RETURN (a) $$) AS (result agtype);
+ERROR:  variable `b` does not exist
+LINE 2:              WHERE EXISTS {(a:person)-[]->(b:pet)}
+                                                   ^
+--query inside
+SELECT * FROM cypher('exists_subquery', $$ MATCH (a:person)
+										   WHERE EXISTS {MATCH (a:person)-[]->(b:pet) RETURN b}
+										   RETURN (a) $$) AS (result agtype);
+                                              result
```
[the remark this answers is not in the document]
Comment (dehowef) [reply regress/expected/cypher_subquery.out:57]: @jrgemignani Updated the Regression tests with messages in the comments, let me know what you think

**Review by jrgemignani — APPROVED** [review]

**Review by MuhammadTahaNaveed — APPROVED** [review]
lgtm!

File: src/backend/parser/cypher_gram.y [hunk 2]
```diff
@@ -594,6 +599,53 @@ updating_clause:
...
+            $$ = $1;
+        }
+    | subquery_stmt_no_return UNION all_or_distinct subquery_stmt_no_return
+        {
+            $$ = list_make1(make_set_op(SETOP_UNION, $3, $1, $4));
+        }
+    ;
+
+single_subquery:
+        subquery_part_init reading_clause_list return
```
Comment (Zainab-Saad) [inline src/backend/parser/cypher_gram.y:636]: extra indentation?

File: src/backend/parser/cypher_gram.y [hunk 3]
```diff
@@ -594,6 +599,53 @@ updating_clause:
...
+        {
+            $$ = list_concat($1, lappend($2, $3));  
+        }
+    ;
+
+single_subquery_no_return:
+        subquery_part_init reading_clause_list
```
Comment (Zainab-Saad) [inline src/backend/parser/cypher_gram.y:643]: extra indentation?

**Review by Zainab-Saad — APPROVED** [review]
Added comments for the formatting changes, otherwise looks good!

File: src/backend/parser/cypher_gram.y (hunk 3 above)
Comment (dehowef) [reply src/backend/parser/cypher_gram.y]: Thanks for pointing that out~ I amended the commit. Feel free to merge it whenever you see fit

**Pull request merged**